### PR TITLE
Bye bye 3rd party

### PR DESCRIPTION
**Pull request by nirvn**
QField now uses QGIS' QgsTessellator class instead of a 3rdparty library. This allows us to drop over 6,000 lines of code in favor of a class we already ship in QGIS' core library.

This also fixes https://github.com/opengisch/QField/issues/1614 :
image

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 5ca5de10c9efba5699ce2f56e0b07a1812156f77:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2302-d7d3a280a8e541e6cb6d0b1a75b66b13bce4f194-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2302-d7d3a280a8e541e6cb6d0b1a75b66b13bce4f194-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2302-d7d3a280a8e541e6cb6d0b1a75b66b13bce4f194-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2302-d7d3a280a8e541e6cb6d0b1a75b66b13bce4f194-x86.apk)

File: src/core/sgrubberband.cpp [hunk 1]
```diff
@@ -73,39 +78,36 @@ QSGGeometryNode *SGRubberband::createLineGeometry( const QVector<QgsPoint> &poin
...
-    vertices_in[i++] = pt.y();
-  }
-
-  tessellate( &coordinates_out, &nverts,
-              &tris_out, &ntris,
-              contours_array, contours_array + 2 );
+  QgsPolygon *polygon = new QgsPolygon( new QgsLineString( points ) );
+  QgsTessellator t( 0, 0, false, false, false, true );
+  if ( points.size() > 2 )
+    t.addPolygon( *polygon, 0 );
```
Comment (m-kuhn) [inline src/core/sgrubberband.cpp:84]:
```suggestion
  QgsPolygon polygon = QgsPolygon( new QgsLineString( points ) );
  QgsTessellator t( 0, 0, false, false, false, true );
  if ( points.size() > 2 )
    t.addPolygon( polygon, 0 );
```

This avoids an unnecessary copy and doesn't leak with points.size() <= 2

**Review by m-kuhn — APPROVED** [review]
Nice, nice, nice ...

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 3d1ff98689c4b7c0a30fed2df4bb034ea33d024c:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2302-1c0b8d66225edc10209e344a4c676d6a270d08e3-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2302-1c0b8d66225edc10209e344a4c676d6a270d08e3-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2302-1c0b8d66225edc10209e344a4c676d6a270d08e3-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2302-1c0b8d66225edc10209e344a4c676d6a270d08e3-x86.apk)

**Pull request merged**